### ISSUE TITLE
[Submission] 140 params - Dimo Pepelyashev

### ISSUE THREAD

**Issue by dimopep** [anadim/AdderBoard#81]
### Author

Dimo Pepelyashev

### Unique Parameter Count

140

### Accuracy

100.00% (10,010/10,010)

### Method

Trained (SGD/Adam)

### Architecture

1L decoder, d=4, 1h/1kv, hd=4, ff=4, RoPE θ=3, SwiGLU 

### Key Tricks

 Tied K=V, tied O=Q^T, tied lm_head, QK-norm

### Link to Code

https://gist.github.com/dimopep/27158a2b0ed983e32ee8f39af6e5a134

### Verification Output

```shell
Model: 140p Adder (d=4, tied KV/OQ/LM, QK-norm)
Author: Dimo Pepelyashev
Parameters (unique): 140
Architecture: 1L decoder, d=4, 1h/1kv, hd=4, ff=4, RoPE theta=3, SwiGLU
Tricks: tied K=V, tied O=Q^T, tied lm_head, QK-norm, autoregressive teacher-forced loss

  Progress: 1000/10010 (1000/1000 correct) [2.8s]
  Progress: 2000/10010 (2000/2000 correct) [5.7s]
  Progress: 3000/10010 (3000/3000 correct) [8.5s]
  Progress: 4000/10010 (4000/4000 correct) [11.3s]
  Progress: 5000/10010 (5000/5000 correct) [14.1s]
  Progress: 6000/10010 (6000/6000 correct) [16.9s]
  Progress: 7000/10010 (7000/7000 correct) [19.7s]
  Progress: 8000/10010 (8000/8000 correct) [22.5s]
  Progress: 9000/10010 (9000/9000 correct) [25.3s]
  Progress: 10000/10010 (10000/10000 correct) [28.3s]

Results: 10010/10010 correct (100.00%)
Time: 28.4s (353 additions/sec)
Status: QUALIFIED (threshold: 99%)
```

## Training

- **Optimizer:** AdamW, lr=0.01, no weight decay
- **Batch size:** 128
- **Steps:** 20,000 (converged to 100% eval accuracy)
- **Loss:** Autoregressive teacher-forced (11 forward passes per step, feeding ground truth tokens)
- **Curriculum:** None (full 10-digit random pairs from step 1)
- **Hardware:** Single NVIDIA A100-SXM4-80GB (~17 minutes)

## Exploration 

- Claude Code (Opus 4.6) w/  8x A100 SXM
